catalog_full.py

```diff
@@ -0,0 +1,58 @@
+import os
+import urllib
+import jinja2
+import webapp2
+import logging
+import yaml
+
+import models
+import authorizer
+import schemas
+
+JINJA_ENVIRONMENT = jinja2.Environment(
+    loader=jinja2.FileSystemLoader(os.path.dirname(__file__)),
+    extensions=['jinja2.ext.autoescape'],
+    autoescape=True)
+
+class CatalogFull(webapp2.RequestHandler):
+  def SortByName(self, classes):
+    if not classes:
+      return ''
+    return sorted(classes, key=lambda e: e['name'])
+
+  def CoreClasses(self, classes):
+    return [c for c in classes if 'Core' in c['name']]
+
+  def get(self):
+    auth = authorizer.Authorizer(self)
+    if not auth.HasStudentAccess():
+      auth.Redirect()
+      return
+    
+    institution = self.request.get("institution")
+    if not institution:
+      logging.fatal("no institution")
+    session = self.request.get("session")
+    if not session:
+      logging.fatal("no session")
+    if not auth.HasPageAccess(institution, session, "materials"):
```

Comment:
So on this page this might not be required. We want this to be available on all serving levels right? On other pages, like the schedule page, we might write something like this:

```python
    if not auth.ServingLevel(institution, session) == "schedule":
      auth.Redirect()
      return
```